src/core/sock.c

```diff
@@ -313,6 +314,11 @@ static int nn_sock_setopt_inner (struct nn_sock *self, int level,
                 return -EINVAL;
             dst = &self->rcvbuf;
             break;
+        case NN_RCVMAXSIZE:
+            if (nn_slow (val <= -1))
```

Comment:
Seems that way. What would expected behavior be? Any integer <=-1 disables? Or only -1 disables, and <=-2 returns EINVAL? I don't have strong opinions, except to choose the lesser surprising option.